### ISSUE TITLE
Popups Vanishing

### ISSUE THREAD

**Issue by DanTheMan1244** [elParaguayo/qtile-extras#91]
Whenever I make a Popup that has focusable widgets, it will disappear whenever I move my mouse off of the popup. It will also disappear if a new window appears. This never happens if no controls can be focused.

This is the code I have used to define my Popup:
```
def startPopups(qtile):
    global PopupLayout
    PopupLayout = PopupRelativeLayout(qtile, width=1000, height=200, controls=[
        PopupWidget(
            widget=widget.CPUGraph(),
            width=0.45,
            height=0.45,
            pos_x=0.05,
            pos_y=0.05,
            highlight_method='border'
        ),
        PopupWidget(
            widget=widget.MemoryGraph(),
            width=0.9,
            height=0.45,
            pos_x=0.05,
            pos_y=0.5,
        ),

    ], background="00000060", inital_focus=None, close_on_click=False)


@lazy.function
def show_graphs(qtile):
    startPopups(qtile)
    PopupLayout.show(centered=True)

@lazy.function
def hide_graphs(qtile):
    PopupLayout.hide()

```
This Popup will stay.

However, If I add a mouse callback, like so:
```
def startPopups(qtile):
    global PopupLayout
    PopupLayout = PopupRelativeLayout(qtile, width=1000, height=200, controls=[
        PopupWidget(
            widget=widget.CPUGraph(),
            width=0.45,
            height=0.45,
            pos_x=0.05,
            pos_y=0.05,
            highlight_method='border'
        ),
        PopupWidget(
            widget=widget.MemoryGraph(),
            width=0.9,
            height=0.45,
            pos_x=0.05,
            pos_y=0.5,
            mouse_callbacks={"Button1": hide_graphs()}

        ),

    ], background="00000060", inital_focus=None, close_on_click=False)


@lazy.function
def show_graphs(qtile):
    startPopups(qtile)
    PopupLayout.show(centered=True)

@lazy.function
def hide_graphs(qtile):
    PopupLayout.hide()

```
This will disappear if I move the mouse off of the Popup, or if a new window appears.
I have noticed while writing this, that the Popup also disappears if the focus of the windows is changed, which if I have the follow_mouse_focus property set as true, in the qtile config, would explain why the popup disappears when my mouse moves off of the Popup. It would also explain why if a new window appears why the popup will also disappear.
Again none of this happens if the popup has no focusable controls.

**Comment by elParaguayo** [comment]
That is the intended behaviour at the moment. When using keyboard navigation the popup subscribes to some hooks which will kill the window if it loses focus. This is the relevant code:
https://github.com/elParaguayo/qtile-extras/blob/159eb19b9e796b7b685c8806c40cca709ed25f59/qtile_extras/popup/toolkit.py#L202-L219

So, we would just need an optional parameter to disable the hooks.

If you comment out line 203, does it work as expected/desired?

**Comment by DanTheMan1244** [comment]
I tried commenting out line 203 as you suggested. It did not have any effect on the behavior of the Popup.

**Comment by elParaguayo** [comment]
Commenting out line 203 fixes the issue for me. Can you try again?

**Comment by DanTheMan1244** [comment]
I have tried again. Simply commenting out line 203 and then writing the file and reloading qtile does not do anything for me. I have also tried commenting out the entire set_hooks() function, but that also did nothing.

**Comment by elParaguayo** [comment]
What do you mean by "reloading"? Are you completely exiting qtile?

**Comment by DanTheMan1244** [comment]
No, just using a keybinding to call lazy.reload_config()

**Comment by elParaguayo** [comment]
That won't work. You'll need to exit qtile to load the new code. You're not changing your config.

**Comment by DanTheMan1244** [comment]
Yes, that worked. I had figured that reloading the config would work due to the popup being defined in the config, but I understand now.

**Comment by elParaguayo** [comment]
Ok, let me play around with this. May see if we can have an option to disable those hooks rather than you having to hard code it.
